### DISCUSSION TITLE
Post new Team response headers

### DISCUSSION THREAD

**Discussion by Maki079** [microsoftgraph/msgraph-sdk-go#637] [Q&A] [answered]
Hi!
Can someone please help me with my struggle ? 
I want to create a new MsTeam and after creation i need team guId. From what i can see using https://developer.microsoft.com/en-us/graph/graph-explorer the newly created team guid is returned in response headers like so :
`{
    "client-request-id": "123XSX-xxxx-xxxx-xxxx-b14c9123121c70",
    "content-length": "0",
    "location": "/teams('ple4se-bb0d-831p-97eb-143xxx123fc8')/operations('a324sda-2137-xxx-xxxx-654dxasfs')",
    "request-id": "123XSX-xxxx-xxxx-xxxx-b14c9123121c70"
}`

but response body is empty.

IN go sdk
when i use 
`
	result , err := graphClient.Teams().Post(context.Background(), requestBody, nil)
`
ofc when the request is successful result will be nill , but is there any way to access response headers in this sdk for this function?

I'm hugely thankful for any helpful response

**Comment by rkodev** [comment, marked as answer]
Hi @Maki079, thanks for highlighting such scenarios.
In cases where you would want to access the response header, you will need to define a response handler and define your serialization response. 

Here is an example of the code

```go

	// define a response handler
	responseHandler := func(response interface{}, errorMappings abstractions.ErrorMappings) (interface{}, error) {
		// cast response to the type you expect
		httpRes := response.(*http.Response)
		// read the headers
		fmt.Printf("Headers: %v\n", httpRes.Header)

		return nil, nil
	}

	// create a context with the response handler
	ctx := context.Background()
	ctx = context.WithValue(ctx, abstractions.ResponseHandlerOptionKey, responseHandler)

	// execute the request
	result, err := client.Teams().Post(ctx, requestBody, nil)

```
